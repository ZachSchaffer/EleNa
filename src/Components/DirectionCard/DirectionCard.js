import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

export default class DirectionCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currLat: this.props.currLat,
      currLong: this.props.currLong,
      destLat: this.props.destLat,
      destLong: this.props.destLong,
    };

    this.getDirection = this.getDirection.bind(this);
    this.getDistance = this.getDistance.bind(this);
  }

  getDirection() {
    return '6° East';
  }

  getDistance() {
    return '2 Miles';
  }

  render() {
    return (
      <Card style={{ minWidth: '20vw' }}>
        <CardContent>
          <Typography color="textSecondary" gutterBottom>
            Current Location: 30.438493° West 10° North
          </Typography>
          <hr />
          <Typography color="textSecondary" gutterBottom>
            Destination: Boston, MA
          </Typography>
          <Typography variant="h5" component="p">
            Turn {this.getDirection()}
          </Typography>
          <Typography variant="h5" component="p">
            Travel {this.getDistance()}
          </Typography>
        </CardContent>
        <CardActions>
          <Button size="small">Next Direction</Button>
        </CardActions>
      </Card>
    );
  }
}
