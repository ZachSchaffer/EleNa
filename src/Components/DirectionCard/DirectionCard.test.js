import React from 'react';
import renderer from 'react-test-renderer';
import DirectionCard from './DirectionCard';

test('test DirectionCard', () => {
    const tree = renderer.create(<DirectionCard />).toJSON();
    expect(tree).toMatchSnapshot();
});