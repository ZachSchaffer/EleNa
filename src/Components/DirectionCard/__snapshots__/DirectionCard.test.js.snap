// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DirectionCard 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "minWidth": "20vw",
    }
  }
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
    >
      Current Location: 30.438493° West 10° North
    </p>
    <hr />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
    >
      Destination: Boston, MA
    </p>
    <p
      className="MuiTypography-root MuiTypography-h5"
    >
      Turn 
      6° East
    </p>
    <p
      className="MuiTypography-root MuiTypography-h5"
    >
      Travel 
      2 Miles
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Next Direction
      </span>
    </button>
  </div>
</div>
`;
